Cache/Requests.lean

```diff
@@ -161,25 +159,26 @@ end Get
 
 section Put
 
+/-- FRO cache S3 URL -/
+def UPLOAD_URL : String :=
+  "https://a09a7664adc082e00f294ac190827820.r2.cloudflarestorage.com/mathlib4"
+
 /-- Formats the config file for `curl`, containing the list of files to be uploaded -/
-def mkPutConfigContent (fileNames : Array String) (token : String) : IO String := do
+def mkPutConfigContent (fileNames : Array String) : IO String := do
   let l ← fileNames.data.mapM fun fileName : String => do
-    pure s!"-T {(IO.CACHEDIR / fileName).toString}\nurl = {← mkFileURL fileName (some token)}"
+    pure s!"-T {(IO.CACHEDIR / fileName).toString}\nurl = {UPLOAD_URL}/f/{fileName}"
   return "\n".intercalate l
 
 /-- Calls `curl` to send a set of cached files to the server -/
 def putFiles (fileNames : Array String) (overwrite : Bool) (token : String) : IO Unit := do
+  -- TODO: reimplement using HEAD requests?
+  let _ := overwrite
```

Comment:
I am relatively sure this is not critical, but do we want it back at all? There could be a slight benefit of not paying for more costly write operations, but the number of `cache put` calls should not matter compared to the many more `cache get` calls.

Reply:
To explain further, the S3 protocol as a distributed storage protocol will *always* override existing files

Reply:
I think we do want to avoid overwriting files if possible. The replacement files will never be the same as the ones that are there already because they contain metadata about the commit on which the CI was run, so this will repeatedly overwrite the files, meaning that there is less room for network caching effects.

Reply:
Oh, where is that metadata? I'm only aware of the `/c/{githash}` files but those are conflict-free

Reply:
It's in the ltar files themselves, added via a command line option in the `leantar` invocation in cache. I added it because it's useful for debugging when you receive a cache file from somewhere which has something weird in it, to track down the CI run that produced it for anomalies.

Reply:
I think we could also avoid a second request by logging this information when we do `lake exe cache get` at the beginning of the run

Reply:
Yes. We might even be able to go with mtimes for the sake of simplicity - upload every .olean file newer than the start of the build

Reply:
Well we're uploading .ltar files, not .olean files... but I think `cache put` already has the information needed there, since it calls `cache pack` which does not replace existing cache files. The trouble is that there might be a cache file that exists locally on the CI machine, which was not cleared by `cache clean` because it is useful to skip part of the build, but does not exist in the cloud (somehow); in this case it would see that the file already existed previously and would not upload it, meaning that the file would still not exist in the cloud afterwards. Ideally the system should be able to recover from situations like that.